controller/utils.go

```diff
@@ -276,51 +271,83 @@ func waitTerminatingPods(
 		exit := true
 		select {
 		case <-ticker.C:
-			for _, pod := range deletedPods {
+			if operationManager.WasCanceled() {
```

Comment:
shouldn't you have the same check here?
`if operationManager != nil && operationManager.WasCanceled() {`

Reply:
Yes it should

Reply:
Updated